.github/workflows/ci.yaml

```
@@ -88,6 +88,16 @@ jobs:
       - name: Run diff against branch
         run: scripts/diff_against_branch.sh
 
+  # diff check against dev branch to help with the PR reviews
+  diff-dev:
+    name: Diff against dev to help with the PR reviews
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v4
+      - name: Run diff against branch
+        run: scripts/diff_against_branch.sh dev || true && echo "\n🛟  THIS IS ONLY INFORMATIONAL. ERRORS ARE NOT VALID FOR DEV BRANCH."
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2028:info:1:52: echo may not expand escape sequences. Use printf

(shellcheck)